packages/queued-request-controller/tests/utils.ts

```diff
@@ -13,7 +15,7 @@ import type {
  */
 export function buildMessenger() {
```

Comment:
It seems that we are sharing this messenger across the tests for both the controller and the middleware. While that does reflect how the controller and middleware will be used in practice, I suppose, I don't know that we really need to do this.

I feel like each test file ought to have its own function for building a messenger. We can add this function to the bottom of the controller test file:

``` typescript
function buildQueuedRequestControllerMessenger(): QueuedRequestControllerMessenger {
  const messenger = new ControllerMessenger();
  return messenger.getRestricted({
    name: 'QueuedRequestController',
  });
}
```

and then we can add this function to the bottom of the middleware test file:

``` typescript
function buildQueuedRequestMiddlewareMessenger(): QueuedRequestMiddlewareMessenger {
  return new ControllerMessenger<
    QueuedRequestControllerActions | AllowedActions,
    QueuedRequestControllerEvents
  >();
}
```

and that way we don't even need this file.

(This builds upon the comment I made earlier, so now the controller messenger doesn't need to allow anything.)

What are your thoughts?

Reply:
Applied here dc3ca45810a07647f13b96484838d45e9f444c26! I also applied the same changes to `selected-network-controller`, which is the only other controller that defined build messenger functions in a separate file: b6494281b91ecbafe987679edc055cf4a8a615d3.

I don't think these changes are strictly necessary. The functions return a new messenger instance with its own closure scope every time they're invoked, so it shouldn't matter whether or not the test-messenger template/factory is located in the same file as the tests that are invoking the factory (not messenger instance). 

(i.e. The `QueuedRequestMiddleware` tests all share a single instance of `QueuedRequestMiddlewareMessenger`, while the `QueuedRequestController` tests all use separate instances of `QueuedRequestControllerMessenger`, but isn't affected by whether or not the `build.+Messenger()` function definitions are located in the same test file or in `tests/utils.ts`.)

If anything, it seems to be that once the same messenger type starts being used in multiple test files, moving the factory functions to a separate file will become necessary. But I'm also good with crossing that bridge when we reach it, and reverting this as a premature optimization for now.